Unhandled response: RequestLimitExceeded

My first test run against my production AWS account resulted in panic. AWS limits requests, but it's an hard thing to hit. You might want to check/limit this somehow. Or handle the error response properly. 

```
panic: RequestLimitExceeded: Request limit exceeded.
	status code: 503, request id: c8d70a14-7ee6-4110-9772-d7aaab8a6a25

goroutine 1 [running]:
main.isInstanceInPublicSubnet(0xc0000a00b8, 0xc0007e6680, 0x17a4ba9)
	/Users/margus/src/go/src/github.com/danihodovic/generate-ssh-configs/aws.go:155 +0x4f9
main.generateAWS(0xc0000968a7, 0x5)
	/Users/margus/src/go/src/github.com/danihodovic/generate-ssh-configs/aws.go:285 +0x513
main.glob..func2(0x1dbbde0, 0xc0000a6d20, 0x0, 0x2)
	/Users/margus/src/go/src/github.com/danihodovic/generate-ssh-configs/cmd.go:29 +0x47
github.com/spf13/cobra.(*Command).execute(0x1dbbde0, 0xc0000a6d00, 0x2, 0x2, 0x1dbbde0, 0xc0000a6d00)
	/Users/margus/src/go/src/github.com/spf13/cobra/command.go:766 +0x2cc
github.com/spf13/cobra.(*Command).ExecuteC(0x1dbbb80, 0x14, 0x1dbc040, 0x1dbbde0)
	/Users/margus/src/go/src/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/spf13/cobra.(*Command).Execute(0x1dbbb80, 0x0, 0x17a60c6)
	/Users/margus/src/go/src/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
	/Users/margus/src/go/src/github.com/danihodovic/generate-ssh-configs/main.go:16 +0x36
```

--- a/README.md
+++ b/README.md
@@ -108,6 +108,7 @@ bindkey "^s" fzf-ssh
   - Uses the public IP if
     - the instance is in a public subnet
     - the security group allows ingress port 22 from the public internet
+    - the security group allows ingress port 22 from subnet provided via `--subnet` flag
   - Otherwise it uses the private IP and routes through the jumphost if one is
     configured.
 - **DigitalOcean**

--- a/aws.go
+++ b/aws.go
@@ -179,10 +179,15 @@ func isPortOpen(ec2Client *ec2.EC2, inst *ec2.Instance) bool {
 		panic(err)
 	}
 
+	publicSubnets := []string{"0.0.0.0/0"}
+	if openSubnet != "" {
+		publicSubnets = append(publicSubnets, openSubnet)
+	}
+
 	for _, securityGroup := range securityGroupOutput.SecurityGroups {
 		for _, permission := range securityGroup.IpPermissions {
 			protocolOk := (*permission.IpProtocol == "tcp" || *permission.IpProtocol == "-1")
-			publicIpRanges := usesPublicIpRanges(permission.IpRanges)
+			publicIpRanges := usesPublicIpRanges(permission.IpRanges, publicSubnets)
 
 			if permission.FromPort == nil && permission.ToPort == nil &&
 				publicIpRanges &&
@@ -203,10 +208,12 @@ func isPortOpen(ec2Client *ec2.EC2, inst *ec2.Instance) bool {
 	return false
 }
 
-func usesPublicIpRanges(ipRanges []*ec2.IpRange) bool {
+func usesPublicIpRanges(ipRanges []*ec2.IpRange, subnets []string) bool {
 	for _, r := range ipRanges {
-		if *r.CidrIp == "0.0.0.0/0" {
-			return true
+		for _, s := range subnets {
+			if *r.CidrIp == s {
+				return true
+			}
 		}
 	}
 

--- a/cmd.go
+++ b/cmd.go
@@ -11,6 +11,7 @@ var sshUser string
 var identityFile string
 var jumphost string
 var filters string
+var openSubnet string
 
 var rootCmd = &cobra.Command{
 	Use: "generate-ssh-configs",
@@ -90,6 +91,12 @@ func cmdInit() {
 		"filters",
 		"",
 		"AWS instance filters",
+	)
+	awsCmd.Flags().StringVar(
+		&openSubnet,
+		"subnet",
+		"",
+		"Additional open subnet",
 	)
 
 	rootCmd.AddCommand(awsCmd)

